src/core/dep-graph.ts

```diff
@@ -327,26 +303,41 @@ class DepGraphImpl implements types.DepGraphInternal {
     return node;
   }
 
-  private pathsFromNodeToRoot(nodeId: string): types.PkgInfo[][] {
+  private pathsFromNodeToRoot(
+    nodeId: string,
+    ancestors: string[] = [],
```

Comment:
Are you very sure you don't want this to be a Set? (isn't `Set.has` faster than `Array.includes` for large sets?)